moment_kinetics/src/input_structs.jl

```diff
@@ -733,4 +784,27 @@ function Dict_to_NamedTuple(d)
     return NamedTuple(Symbol(k)=>v for (k,v) ∈ d)
 end
 
+"""
+Dict merge function for named keyword arguments 
+for case when input Dict is a mixed Dict of Dicts
+and non-Dict float/int/string entries, and the 
+keyword arguments are also a mix of Dicts and non-Dicts
+"""
+
+function merge_dict_with_kwargs!(dict_base; args...)
+    #println("before merge: ",dict_base)
+    for (k,v) in args
+        println(k, " ", v)
+        if String(k) in keys(dict_base)
+            if isa(v,AbstractDict)
+                v = merge(dict_base[String(k)],v)
+            end            
+        end
+        dict_mod = Dict(String(k) => v)
+        dict_base = merge(dict_base, dict_mod)
+    end
+    #println("after merge: ",dict_base)
```

Comment:
This change assumes that all Dicts are declared as Dict{String,Any}. I don't think that the case for now, so if you agree, I'll add the OptionsDict type first so that `dict_base[k] = v` is always valid.